velox/connectors/hive/SplitReader.cpp

```diff
@@ -91,7 +92,19 @@ void SplitReader::prepareSplit(
   VELOX_CHECK_NE(
       baseReaderOpts_.getFileFormat(), dwio::common::FileFormat::UNKNOWN);
 
-  auto fileHandle = fileHandleFactory_->generate(hiveSplit_->filePath).second;
+  std::shared_ptr<FileHandle> fileHandle = nullptr;
+  try {
+    fileHandle = fileHandleFactory_->generate(hiveSplit_->filePath).second;
+  } catch (VeloxRuntimeError& e) {
```

Comment:
I have seen some discussions about the poor performance of try-catch, and am not sure if there's better way to achieve this. Maybe [Status](https://github.com/facebookincubator/velox/blob/main/velox/common/base/Status.h) is an option. Thanks.

Reply:
@rui-mo Throwing is problematic when it happens per-row. Here, throwing should be fine.